### PR TITLE
Lesson6 Homework

### PR DESCRIPTION
**Pull request by NetMic**

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
this also includes lesson 5 which was closed: 1285

File: students/Net_Michael/lesson5/mailroom3.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        list_files = os.listdir(donor_folder_new)
+        folder_new = os.path.basename(donor_folder_new)
+        if(len(list_files) > 0):
+            while True:
+                new_donor_folder = folder_new
+                sel_choice = eval(input(prompt))
+                if sel_choice < 6:
+                    main_sel.get(sel_choice)()
+                else:
+                    print("\n choose less than 4 or press 4 to exit \n")
```
Comment (charlesrobison) [inline students/Net_Michael/lesson5/mailroom3.py]: @NetMic - unable to run this due to UnboundLocalError.  I'll submit partial credit.  Please submit in a new PR when this is corrected.  Reach out if you have questions.

File: students/Net_Michael/lesson6/mailroom4.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+# given the naame of the folder containing the list of donors
+# such as Paul_Allen.txt, the function provides the path of the folder from
+# any machine
+
+def directory_path(donor_folder = new_donor_folder):
+    dir_info = list(list_dir_path())
+    for j in range(len(dir_info)):
+        if dir_info[j][0] in donor_folder:
+            sel_dirct = dir_info[j][1]
+    donation_directory = os.path.join(sel_dirct, donor_folder)
```
Comment (charlesrobison) [inline students/Net_Michael/lesson6/mailroom4.py]: Getting UnboundLocalError on this file as well.

File: students/Net_Michael/lesson6/mailroom4.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        list_files = os.listdir(donor_folder_new)
+        folder_new = os.path.basename(donor_folder_new)
+        if(len(list_files) > 0):
+            while True:
+                new_donor_folder = folder_new
+                sel_choice = eval(input(prompt))
+                if sel_choice < 6:
+                    main_sel.get(sel_choice)()
+                else:
+                    print("\n choose less than 4 or press 4 to exit \n")
```
Comment (charlesrobison) [inline students/Net_Michael/lesson6/mailroom4.py]: @NetMic - getting the same error in this code as well.  Please submit corrected code in a new PR.

File: students/Net_Michael/lesson7/html_render.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    tag = 'ul'
+
+class Li(Element):
+    tag = 'li'
+
+class H(OneLineTag):
+    def __init__(self, size, content, **kwargs):
+        OneLineTag.__init__(self, content, **kwargs)
+        self.size = size
+        self.tag = "h{}".format(size)
```
Comment (charlesrobison) [inline students/Net_Michael/lesson7/html_render.py]: Looks great!

File: students/Net_Michael/lesson8/circle.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        return diameter
+        #
+    @diameter.setter
+    def diameter(self, dd):
+        self._radius = dd/2
+    #
+    @property
+    def area(self):
+        area = math.pi * self._radius **2
+        return area
```
Comment (charlesrobison) [inline students/Net_Michael/lesson8/circle.py]: Nice job on this!

**Pull request merged**